In [35]:
import pandas as pd
import numpy as np
import pingouin as pg

TASK 1: NHẬP KHẨU DỮ LIỆU

In [36]:
path = 'data_python.csv'
dataframe = pd.read_csv(path)
df = dataframe
df.head()

Ho_va_ten       So_dien_thoai_hoac_email Gioi_tinh       Do_tuoi  \
0          Kieu Anh  ngothikieuanh281002@gmail.com        Nu       Duoi 30   
1     Dang Bao Ngan                      708631096        Nu       Duoi 30   
2  Tran Cao Nam Anh      trancaonamanh93@gmail.com        Nu       Duoi 30   
3  Nguyen Cong Hung                      981163301       Nam  Tu 30 den 39   
4           Ho Dieu                      325791256        Nu       Duoi 30   

     Tham_nien_cong_tac      Noi_lam_viec     Nganh                 Chuc_vu  \
0   Tu 1 den duoi 3 nam  Tap doan Hoa Sen  Ton thep               Nhan vien   
1   Tu 3 den duoi 5 nam      Thep Nam Kim  Ton thep               Nhan vien   
2   Tu 3 den duoi 5 nam  Tap doan Hoa Sen  Ton thep               Nhan vien   
3  Tu 5 den duoi 10 nam  Tap doan Hoa Sen  Ton thep  Giam doc/ Pho giam doc   
4   Tu 1 den duoi 3 nam        Ton Dong A  Ton thep               Nhan vien   

     Bo_phan Loai_hinh_doanh_nghiep  ... TT4 GS1  GS2  GS2.1  GS4  HH1  HH2  \
0    Ke toan        Cong ty co phan  ...   5   5    4      3    3    5    5   
1  Tai chinh        Cong ty co phan  ...   3   5    4      4    3    5    4   
2    Ke toan        Cong ty co phan  ...   4   4    4      5    5    2    5   
3   Mua hang        Cong ty co phan  ...   5   5    5      5    4    4    3   
4    Ke toan        Cong ty co phan  ...   3   4    4      3    4    2    3   

   HH3  HH4  HH5  
0    5    4    3  
1    4    5    5  
2    3    4    3  
3    5    5    4  
4    5    4    5  

[5 rows x 58 columns]

TASK 2: LÀM SẠCH DỮ LIỆU

In [37]:
#2.1 Check Null Values

df.isnull().sum()

Ho_va_ten                    0
So_dien_thoai_hoac_email    17
Gioi_tinh                    0
Do_tuoi                      0
Tham_nien_cong_tac           0
Noi_lam_viec                 0
Nganh                        0
Chuc_vu                      0
Bo_phan                      0
Loai_hinh_doanh_nghiep       0
Linh_vuc_hoat_dong           0
Quy_mo                       0
MT1                          0
MT2                          0
MT3                          0
MT4                          0
MT5                          0
MT6                          0
MT7                          0
MT8                          0
MT9                          0
MT10                         0
MT11                         0
MT12                         0
MT13                         0
MT14                         0
RR1                          0
RR2                          0
RR3                          0
RR4                          0
RR5                          0
RR6                          0
KS1     

In [38]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91 entries, 0 to 90
Data columns (total 58 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Ho_va_ten                 91 non-null     object
 1   So_dien_thoai_hoac_email  74 non-null     object
 2   Gioi_tinh                 91 non-null     object
 3   Do_tuoi                   91 non-null     object
 4   Tham_nien_cong_tac        91 non-null     object
 5   Noi_lam_viec              91 non-null     object
 6   Nganh                     91 non-null     object
 7   Chuc_vu                   91 non-null     object
 8   Bo_phan                   91 non-null     object
 9   Loai_hinh_doanh_nghiep    91 non-null     object
 10  Linh_vuc_hoat_dong        91 non-null     object
 11  Quy_mo                    91 non-null     object
 12  MT1                       91 non-null     int64 
 13  MT2                       91 non-null     int64 
 14  MT3                       91

--> Do đó, dữ liệu không có khoản trắng. Ngoại từ cột "So_dien_thoai_hoac_email", trường dữ liệu cột này không quan trọng cho nghiên cứu, do đó không nhất thiết phải đầy đủ.

In [39]:
##2.2 Check values of all columns

column_headers = list(df.columns.values)
n = len(df.columns)
for i in range(n):
    print(i, end = " ")
    print('- ', end = " ")
    print(column_headers[i], end = " ")
    print(set(df[column_headers[i]]))

0 -  Ho_va_ten {'Nguyen Minh Quan', 'Nguyen Minh Hien', 'Pham Van Dinh', 'Nguyen Hoang Duc', 'Phan Thi Ngoc Chi', 'Nguyen Thi Cam Ty', 'Nguyen Hoang To Loan', 'Luu Thi My Thanh', 'Tran Dinh Nam', 'Nguyen Thai Khanh Huyen', 'Do Thi Thanh Huyen', 'Cao Phuong Mai', 'Nguyen Thi Loan ', 'Tran Van Bao', 'Nguyen Thi Thanh Tam', 'Nguyen Thi Ha Phuong', 'Tran Dang Khoa', 'Nguyen Truc Phuong', 'Nguyen Van An', 'Tran Thi Thu Quyen ', 'Nguyen Thanh Tung', 'Tran Thanh Do', 'Nguyen Ngoc Tran', 'Le Thi Bao Ngoc', 'Nguyen Thi Kim Thu', 'Nguyen Quoc Nhi Bac', 'Le Thi Ngoc Han ', 'Ngo Minh Tri', 'Doan Thi My Ngan', 'Dang Thuy Duong', 'Bui Nhan Tran', 'Vu Thi Minh Phuong ', 'Le Thi Lam', 'Nguyen Van Thi', 'Hoang Mai Anh Dung', 'Phan Tien Cuong', 'Nguyen Thi Thanh Trang', 'Pham Thi Thuy Nhung', 'Nguyen Tu Minh', 'Dang Hai Son', 'Nguyen Thi Thanh Ha', 'Phan Tam', 'Phan Dieu', 'Nguyen Cong Hung', 'Do Thi Mai Xuon', 'Nguyen Le Gia Han', 'Pham Thi Minh Thu', 'Le Thi Ngoc Tram', 'Nguyen Thi Thuy Duong ', 'Ho T

Lọc dữ liệu nghiên cứu:
+ Môi trường (đang công tác hoặc đã công tác)
    Một trong các môi trường sau:
        1.1.	Công ty sản xuất Tôn
        1.2.	Công ty dịch vụ kiểm toán/ kế toán/ tư vấn (Có thực hiện dịch vụ kiểm toán/ kế toán/ tư vấn công ty sản xuất Tôn)
        1.3.	Trường đại học/ học viện đào tạo khối ngành kinh tế
+ Kinh nghiệm (đang công tác hoặc đã công tác)
    Đáp ứng đồng thời các điều kiện kinh nghiệm sau:
        2.	Ít nhất đã hoặc đang làm/ giảng dạy một trong các lĩnh vực về: kế toán, tài chính, kiểm toán, kiểm soát nội bộ, bán hàng, sản xuất, nhân sự, tổng hợp, quản lý
        3.	Ít nhất 1 năm kinh nghiệm trong lĩnh vực thuộc mục (2) và trong môi trường thuộc mục (1)
        4.	Chức danh công việc từ nhân viên trở lên

2.2a Linh_vuc_hoat_dong Column

Chỉ chấp nhận công ty sản xuất, công ty dịch vụ kiểm toán/ kế toán/ tư vấn và các trường đại học/ học viện đào tạo khối ngành kinh tế
    + San xuat
    + Giao duc

In [40]:
print(set(df['Linh_vuc_hoat_dong']))

{'Gia cong', 'Giao duc', 'San xuat', 'Thuong mai'}


In [41]:
## Thống kê số lượng các phần tử có trong cột

count = df['Linh_vuc_hoat_dong'].value_counts()
print(count)

San xuat      82
Thuong mai     5
Giao duc       2
Gia cong       2
Name: Linh_vuc_hoat_dong, dtype: int64


In [42]:
## Xóa các dòng có 'Gia cong', 'Thuong mai'

df = df[df['Linh_vuc_hoat_dong'] != 'Gia cong']
df = df[df['Linh_vuc_hoat_dong'] != 'Thuong mai']
df

Ho_va_ten       So_dien_thoai_hoac_email Gioi_tinh       Do_tuoi  \
0           Kieu Anh  ngothikieuanh281002@gmail.com        Nu       Duoi 30   
1      Dang Bao Ngan                      708631096        Nu       Duoi 30   
2   Tran Cao Nam Anh      trancaonamanh93@gmail.com        Nu       Duoi 30   
3   Nguyen Cong Hung                      981163301       Nam  Tu 30 den 39   
4            Ho Dieu                      325791256        Nu       Duoi 30   
..               ...                            ...       ...           ...   
86     Pham Van Dinh                            NaN       Nam       Duoi 30   
87      Tran Van Bao                      923713103       Nam       Duoi 30   
88     Nguyen Van An                      389286105       Nam       Duoi 30   
89  Ho Van Hong Hanh                      366736024        Nu       Duoi 30   
90    Nguyen Van Thi                      976894036       Nam       Duoi 30   

      Tham_nien_cong_tac       Noi_lam_viec     Nganh                 Chuc_vu  \
0    Tu 1 den duoi 3 nam   Tap doan Hoa Sen  Ton thep               Nhan vien   
1    Tu 3 den duoi 5 nam       Thep Nam Kim  Ton thep               Nhan vien   
2    Tu 3 den duoi 5 nam   Tap doan Hoa Sen  Ton thep               Nhan vien   
3   Tu 5 den duoi 10 nam   Tap doan Hoa Sen  Ton thep  Giam doc/ Pho giam doc   
4    Tu 1 den duoi 3 nam         Ton Dong A  Ton thep               Nhan vien   
..                   ...                ...       ...                     ...   
86   Tu 1 den duoi 3 nam       Thep Nam Kim  Ton thep               Nhan vien   
87   Tu 3 den duoi 5 nam   Tap doan Hoa Sen  Ton thep       To truong/ To pho   
88   Tu 3 den duoi 5 nam   Tap doan Hoa Sen  Ton thep               Nhan vien   
89   Tu 3 den duoi 5 nam  Tap doan Hoa Phat  Ton thep               Nhan vien   
90   Tu 3 den duoi 5 nam   Tap doan Hoa Sen  Ton thep               Nhan vien   

      Bo_phan Loai_hinh_doanh_nghiep  ... TT4 GS1  GS2  GS2.1  GS4  HH1  HH2  \
0     Ke toan        Cong ty co phan  ...   5   5    4      3    3    5    5   
1   Tai chinh        Cong ty co phan  ...   3   5    4      4    3    5    4   
2     Ke toan        Cong ty co phan  ...   4   4    4      5    5    2    5   
3    Mua hang        Cong ty co phan  ...   5   5    5      5    4    4    3   
4     Ke toan        Cong ty co phan  ...   3   4    4      3    4    2    3   
..        ...                    ...  ...  ..  ..  ...    ...  ...  ...  ...   
86    Ke toan        Cong ty co phan  ...   5   5    5      5    5    5    4   
87   Ban hang        Cong ty co phan  ...   3   5    3      5    3    4    4   
88    Ke toan        Cong ty co phan  ...   3   3    5      4    4    3    5   
89   Ban hang        Cong ty co phan  ...   5   4    4      5    3    5    5   
90   Ky thuat        Cong ty co phan  ...   5   4    4      5    3    5    4   

    HH3  HH4  HH5  
0     5    4    3  
1     4    5    5  
2     3    4    3  
3     5    5    4  
4     5    4    5  
..  ...  ...  ...  
86    5    4    4  
87    5    3    4  
88    5    4    5  
89    4    3    5  
90    5    4    4  

[84 rows x 58 columns]

In [43]:
## Check đã drop được các phần tử không phải là 'Gia cong', 'Thuong mai' trong cột Linh_vuc_hoat_dong hay chưa

count = df['Linh_vuc_hoat_dong'].value_counts()
print(count)

San xuat    82
Giao duc     2
Name: Linh_vuc_hoat_dong, dtype: int64


2.2b Tham_nien_cong_tac column

Dữ liệu có thâm niên công tác dưới 1 năm sẽ bị loại bỏ

In [44]:
print(set(df['Tham_nien_cong_tac']))

{'Tu 10 nam tro len', 'Tu 1 den duoi 3 nam', 'Tu 3 den duoi 5 nam', 'Tu 5 den duoi 10 nam'}


In [45]:
## Thống kê số lượng các phần tử có trong cột

count = df['Tham_nien_cong_tac'].value_counts()
print(count)

Tu 3 den duoi 5 nam     36
Tu 1 den duoi 3 nam     32
Tu 5 den duoi 10 nam     9
Tu 10 nam tro len        7
Name: Tham_nien_cong_tac, dtype: int64


--> Ta thấy không có thâm niên công tác dưới 1 năm

==> DỮ LIỆU ĐÃ SẠCH VÀ CÓ THỂ TIẾP TỤC SỬ DỤNG CHO PHÂN TÍCH TIẾP THEO

TASK 3: PHÂN TÍCH DỮ LIỆU


Theo kết quả nghiên cứu định tính, ta có mô hình hồi quy đa biến sau:
    HH = β0 + β1 x MT + β2 x RR + β3 x KS + β4 x TT + β5 x GS + e

Trong đó:
    HH	: Tính hữu hiệu (biến phụ thuộc)
    MT	: Môi trường kiểm soát (biến độc lập)
    RR	: Đánh giá rủi ro (biến độc lập)
    KS	: Hoạt động kiểm soát (biến độc lập)
    TT	: Hệ thống thông tin và truyền thông (biến độc lập)
    GS	: Hoạt động giám sát (biến độc lập)
    β0	: Hằng số
    β1, β2, β3, β4, β5	: Hệ số các biến độc lập (hệ số hồi quy)
    e	: Phần dư

In [46]:
## 3.1 Thống kê mô tả các biến

df.describe()

MT1        MT2        MT3        MT4        MT5        MT6  \
count  84.000000  84.000000  84.000000  84.000000  84.000000  84.000000   
mean    3.845238   3.773810   3.785714   3.857143   3.678571   4.273810   
std     0.950511   0.961314   0.892694   0.808730   0.823754   0.811828   
min     2.000000   1.000000   2.000000   2.000000   2.000000   3.000000   
25%     3.000000   3.000000   3.000000   3.000000   3.000000   4.000000   
50%     4.000000   4.000000   4.000000   4.000000   3.500000   4.500000   
75%     5.000000   5.000000   4.000000   4.000000   4.000000   5.000000   
max     5.000000   5.000000   5.000000   5.000000   5.000000   5.000000   

             MT7        MT8        MT9       MT10  ...        TT4        GS1  \
count  84.000000  84.000000  84.000000  84.000000  ...  84.000000  84.000000   
mean    3.857143   4.119048   4.107143   4.142857  ...   4.095238   4.095238   
std     0.958700   0.812623   0.836197   0.823493  ...   0.886567   0.844815   
min     2.000000   3.000000   3.000000   3.000000  ...   3.000000   3.000000   
25%     3.000000   3.000000   3.000000   3.000000  ...   3.000000   3.000000   
50%     4.000000   4.000000   4.000000   4.000000  ...   4.000000   4.000000   
75%     5.000000   5.000000   5.000000   5.000000  ...   5.000000   5.000000   
max     5.000000   5.000000   5.000000   5.000000  ...   5.000000   5.000000   

             GS2      GS2.1        GS4        HH1        HH2        HH3  \
count  84.000000  84.000000  84.000000  84.000000  84.000000  84.000000   
mean    4.059524   4.119048   3.690476   4.107143   4.047619   4.000000   
std     0.855192   0.910511   0.904822   0.931612   0.876806   0.864285   
min     2.000000   2.000000   2.000000   2.000000   3.000000   3.000000   
25%     3.000000   3.000000   3.000000   4.000000   3.000000   3.000000   
50%     4.000000   4.000000   4.000000   4.000000   4.000000   4.000000   
75%     5.000000   5.000000   4.000000   5.000000   5.000000   5.000000   
max     5.000000   5.000000   5.000000   5.000000   5.000000   5.000000   

             HH4        HH5  
count  84.000000  84.000000  
mean    4.023810   4.178571  
std     0.821051   0.866522  
min     3.000000   3.000000  
25%     3.000000   3.000000  
50%     4.000000   4.000000  
75%     5.000000   5.000000  
max     5.000000   5.000000  

[8 rows x 46 columns]

# 3.2 Kiểm định độ tin cậy bằng hệ số Cronbach’s Alpha

Kiểm định độ tin cậy bằng hệ số Cronbach’s Alpha là phép kiểm định mức độ tương quan chặt chẽ giữa các biến quan sát với nhau trong cùng một nhân tố, nghĩa là, Cronbach’s Alpha cho biết các biến quan sát của một nhân tố, biến nào đóng góp vào việc đo lường khái niệm nhân tố, biến nào không.

Điều kiện để xét hệ số Cronbach's Alpha phù hợp là:
    + Theo Nguyễn Đình Thọ (2013) có chỉ ra một số điều kiện lý thuyết:
        - Hệ số Cronbach’s Alpha chỉ đo lường độ tin cậy của thang đo (từ 3 biến quan sát trở lên), không tính được độ tin cậy cho từng biến quan sát;
        - Hệ số Cronbach’s Alpha có giá trị biến thiên trong đoạn từ 0 đến 1. Về lý thuyết, hệ số này càng cao thì thang đo có độ tin cậy càng cao. Tuy nhiên nếu hệ số quá lớn ≥ 0,95 có khả năng có nhiều biến trong thang đo bị trùng lắp.
    + Theo Nunnally (1978), hệ số Cronbach’s Alpha > 0,6 thì thang đo lường đủ điều kiện.

=> Nghĩa là Cronbach’s Alpha cần > 0,6 và < 0,95 thì đạt độ tin cậy.


In [55]:
##Drop các dữ liệu không phải dạng số để tính Cronbach's Alpha

data = df.drop(columns=['Ho_va_ten', 'So_dien_thoai_hoac_email', 'Gioi_tinh', 'Do_tuoi', 'Tham_nien_cong_tac', 'Noi_lam_viec', 'Nganh', 'Chuc_vu', 'Bo_phan', 'Loai_hinh_doanh_nghiep', 'Linh_vuc_hoat_dong', 'Quy_mo'])
data

MT1  MT2  MT3  MT4  MT5  MT6  MT7  MT8  MT9  MT10  ...  TT4  GS1  GS2  \
0     5    5    5    3    3    3    3    5    3     4  ...    5    5    4   
1     4    5    4    4    4    5    4    5    4     3  ...    3    5    4   
2     5    5    4    4    5    5    4    5    5     5  ...    4    4    4   
3     4    4    4    5    5    4    4    4    5     4  ...    5    5    5   
4     5    5    2    4    3    4    3    4    3     4  ...    3    4    4   
..  ...  ...  ...  ...  ...  ...  ...  ...  ...   ...  ...  ...  ...  ...   
86    3    3    4    5    5    5    5    5    5     5  ...    5    5    5   
87    4    4    4    5    3    3    5    4    4     3  ...    3    5    3   
88    3    3    4    3    3    5    3    3    4     5  ...    3    3    5   
89    4    4    2    3    4    4    3    3    3     5  ...    5    4    4   
90    5    5    4    5    5    4    5    5    3     3  ...    5    4    4   

    GS2.1  GS4  HH1  HH2  HH3  HH4  HH5  
0       3    3    5    5    5    4    3  
1       4    3    5    4    4    5    5  
2       5    5    2    5    3    4    3  
3       5    4    4    3    5    5    4  
4       3    4    2    3    5    4    5  
..    ...  ...  ...  ...  ...  ...  ...  
86      5    5    5    4    5    4    4  
87      5    3    4    4    5    3    4  
88      4    4    3    5    5    4    5  
89      5    3    5    5    4    3    5  
90      5    3    5    4    5    4    4  

[84 rows x 46 columns]

In [62]:
data.columns[1]

'MT2'

In [58]:
##Tính Cronbach Alpha
pg.cronbach_alpha(data)

(0.9162703534213439, array([0.889, 0.94 ]))


==> Cronbach's Alpha = 0.916 > 0.65 và < 0.95 cho nên dữ liệu đạt độ tin cậy và có thể sử dụng cho phân tích tiếp theo.

TASK 4: VẼ BIỂU ĐỒ